python/src/transaction.rs

```diff
@@ -273,46 +292,52 @@ impl FromPyObject<'_> for PyLance<RewriteGroup> {
     }
 }
 
-impl ToPyObject for PyLance<&RewriteGroup> {
-    fn to_object(&self, py: Python<'_>) -> PyObject {
+impl<'py> IntoPyObject<'py> for PyLance<&RewriteGroup> {
+    type Target = PyAny;
+    type Output = Bound<'py, Self::Target>;
+    type Error = PyErr;
+
+    fn into_pyobject(self, py: Python<'py>) -> Result<Self::Output, Self::Error> {
         let cls = py
-            .import_bound(intern!(py, "lance"))
+            .import(intern!(py, "lance"))
             .and_then(|module| module.getattr(intern!(py, "LanceTransaction")))
             .and_then(|cls| cls.getattr(intern!(py, "RewriteGroup")))
             .expect("Failed to get RewriteGroup class");
 
-        let old_fragments = export_vec(py, self.0.old_fragments.as_slice());
-        let new_fragments = export_vec(py, self.0.new_fragments.as_slice());
+        let old_fragments = export_vec(py, self.0.old_fragments.as_slice())?;
+        let new_fragments = export_vec(py, self.0.new_fragments.as_slice())?;
 
         cls.call1((old_fragments, new_fragments))
-            .unwrap()
-            .to_object(py)
     }
 }
 
 impl FromPyObject<'_> for PyLance<RewrittenIndex> {
     fn extract_bound(ob: &Bound<'_, PyAny>) -> PyResult<Self> {
-        let old_id: &str = ob.getattr("old_id")?.extract()?;
-        let new_id: &str = ob.getattr("new_id")?.extract()?;
-        let old_id = Uuid::parse_str(old_id)
+        let old_id: String = ob.getattr("old_id")?.extract()?;
+        let new_id: String = ob.getattr("new_id")?.extract()?;
```

Comment:
Why is it that `extract` works here I wonder?  I feel like all the other string `extract` were changed to `downcast`